### PR TITLE
use coroutines for SetRoutesTest

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/Adapters.kt [hunk 1]
```diff
@@ -77,6 +81,22 @@ suspend fun MapboxNavigation.requestRoutes(options: RouteOptions) =
         }
     }
 
+suspend fun MapboxNavigation.setNavigationRoutesSync(
```
Comment (VysotskiVadim) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/Adapters.kt:84]: Usually people call `sync` the one which blocks current thread. Maybe 
```suggestion
suspend fun MapboxNavigation.setNavigationRoutesAsync(
```

or even
```suggestion
suspend fun MapboxNavigation.setNavigationRoutes(
```
?

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt [hunk 2]
```diff
@@ -70,7 +47,7 @@ class SetRoutesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java)
     }
 
     @Test
-    fun set_navigation_routes_successfully() {
+    fun set_navigation_routes_successfully() = runBlocking {
         val mockRoute = MockRoutesProvider.dc_very_short(activity)
         val route = mockRoute.routeResponse.routes()[0]
```
Comment (VysotskiVadim) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt:52]: Oh, I didn't notice in previous PR. Let's use `mapboxNavigation.requestRoutes(routeOptions)
                .getSuccessfulResultOrThrowException().routes`

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt [hunk 3]
```diff
@@ -70,7 +47,7 @@ class SetRoutesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java)
     }
```
[the remark this answers is not in the document]
Comment (dzinad) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt:47]: > I just think it's a good approach to always follow

I don't quite agree about the "always" part. In my case I don't need different mapboxNavigation objects. I need one for every test. And here I can afford creating it in the Before method and not copypaste the creation in every test (even if it's just one line that invokes the `createMapboxNavigation()` method).
If the test is complex where different options are needed, then it of course makes sense to use the factory in every test. Here I see it as an unnecessary complication.
Comparing the possibility of need of different objects in the future and the overhead for now, I'd leave it as-is.

[the remark this answers is not in the document]
Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt:48]: > I'd leave it as-is.

Sure 👍 

I'm not insisting on changing it. I just shared a personal experience. When I have a choose of 2 approaches which require equal effort, I always pick the more flexible one, even if I don't need those flexibility right now. Feel free to ignore this comment if you don't think it's applicable for your case 👍

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt (hunk 2 above)
Comment (dzinad) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt]: Discussed internally, changed the route creation a little bit.

**Review by VysotskiVadim — APPROVED** [review]

**Pull request merged**